monads/src/HW/Eval.hs

```diff
@@ -29,8 +30,45 @@ initialState = MachineState [] M.empty
 -- Execute a single instruction. 
 -- Successful evaluation does not produce any useful result: only the effect of modifying state matters. 
 execInstr :: (Ord v, Show v) => StackInstr v -> MyState (MachineState v) (Either (Error v) ())
-execInstr = undefined 
+execInstr (PushNum n ) = do
+  MachineState stack env <- get
```

Comment:
It's better to only get a part of the state you need, and use `modify` instead of `put`.